aodntools/timeseries_products/gridded_timeseries.py

```diff
@@ -0,0 +1,262 @@
+                                                                                                depth_min=min(depth_bins),
+                                                                                                depth_max = max(depth_bins))})
+    VoI_interpolated.attrs = sort_attributes(VoI_interpolated.attrs)
+
+    ## create the output file name and write the aggregated product as netCDF
+    facility_code = TStools.get_facility_code(file_name)
+    data_code = TStools.get_data_code(VoI) + 'Z'
+    product_type = 'gridded-timeseries'
+    file_version = 2
+    ncout_filename = TStools.generate_netcdf_output_filename(nc=VoI_interpolated, facility_code=facility_code,
+                                                             data_code=data_code, VoI=VoI,
+                                                             site_code=site_code, product_type=product_type,
+                                                             file_version=file_version)
+    ncout_path = os.path.join(base_path, ncout_filename)
+
+    encoding = {'TIME': {'_FillValue': None,
+                         'units': time_units,
+                         'calendar': time_calendar},
+                VoI+'_count': {'_FillValue': None,
+                               'dtype': np.dtype('int')},
+                'DEPTH': {'dtype': np.dtype('float64')},
+                'LONGITUDE': {'_FillValue': False},
+                'LATITUDE': {'_FillValue': False}}
+
+    TStools.write_netCDF_aggfile(VoI_interpolated, ncout_path, encoding)
+
+    return ncout_path
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Gridded time series: interpolate ONE variable from ALL instruments from ALL deployments from ONE site into 1hr timestamps and fixed depth bins")
+    parser.add_argument('-var', dest='varname', help='name of the variable to concatenate. Like TEMP, PSAL', default='TEMP', required=False)
+    parser.add_argument('-file', dest='filename', help='name of the Hourly Time Series Product file that contains the data', default=None, required=False)
+    parser.add_argument('-depth_bins', dest='depth_bins', help='list of depth where the VoI will be interpolated', default=None, nargs='+', required=False)
+    parser.add_argument('-max_separation', dest='max_separation', help='maximum difference between instruments to allow interpolation', default=50, required=False)
+    parser.add_argument('-depth_bins_increment', dest='depth_bins_increment', help='increment in meters for the automatic generated depth bins', default=10, required=False)
+    parser.add_argument('-path', dest='output_path', help='path where the result file will be written. Default ./', default='./', required=False)
+    parser.add_argument('-config', dest='config_file', help='JSON configuration file', default=None, required=False)
+    args = parser.parse_args()
+
+    if args.config_file:
+        with open(config_file) as ff:
```

Comment:
the if logic here is very similar, I would recommend reviewing the key/value names (i.e. using "output_path" or "path" for both cases.

You are also using dict like access and method access, which is just a wart of using different classes/traits - maybe if you read one as dictioanry the block below will be simpler and more elegant.

Reply:
key/values revisited and unified